tasks/fullStack/app/controllers/shops_controller.rb

```diff
@@ -0,0 +1,8 @@
+class ShopsController < ApplicationController
+  def products
+    @products = Shop.find_by(id: params[:shop_id]).products.order(:title)
+    respond_to do |format|
+      format.json { render json: @products }
+    end
+  end
+end
```

Comment:
Good idea to have a separate controller to look up the products.
However, I do not see the benefits in violating REST and introducing `ShopsController#products`.  `ProductController#index` would solve the same purpose while following REST convention. `products` table has a reference to the shop so querying it by the `shop_id` would work just fine

```suggestion
class ProductsController < ApplicationController
  def index
    @products = Product.where(shop_id: params[:shop_id].order(:title)
    respond_to do |format|
      format.json { render json: @products }
    end
  end
end
```

Reply:
hmm this definitely makes more sense, thank you for pointing it out. I think I narrowed in on the idea of a "shops/..." route here without considering whether that made the most sense given the resources being accessed.